### PR TITLE
reworking item groups

### PR DESCRIPTION
**Pull request by SunshineDistillery**
OVERVIEW:
As new items have been added to the game, they've also been added to the item groups in a very haphazard fashion. So item groups right now are messy. Very messy. 

I plan to split up the gigantic item_groups file into a few separate files in their own folder. Along the way I'll try to condense some of the mega-lists to smaller, more legible sizes, and include new functionality that's been added to the game, but which hasn't been applied to (many) of the item groups. Things like varying the number of charges in a flashlight, applying damage to those old motors you find on the side of the road, and especially drawing items from other groups.

CURRENT PROBLEMS:
Item groups with hundreds of items are hard to read. It's even harder to read when you're trying to figure out the probabilities of an item spawning. Also, since it's difficult to guess where or how the item groups are being used, adding an item might make it spawn in an area you never intended.  

Alone that wouldn't be a big deal, but there are also direct effects on game play. Items which are supposed to be rare show up all the time, items which should be common are vanishingly rare simply because they were neglected to be added to the appropriate groups. 

The current item groups also make creating new maptiles a royal pain. For example, if I wanted to make new map tiles for labs, I would have to keep in mind the Lab Challenge when placing items. It would be bad to have science cards or teleporters be too common. But with the existing item groups, it's hard to determine what drop frequencies really are and how to adjust them. Furthermore, if later on someone decides to place Jaquehammers in the Dresser item group, and those items start spawning in lab dorms, that would probably spoil the challenge. So I would have to make own item groups and place them in the mapgen files. But that's bad too, because if each map has its own custom set, that quickly adds up to a lot of groups that are fairly similar and redundant (which is sort of the case now).

FIXES:
I plan to make smaller, more legible, item groups with more specific but generic themes. Those will go in the main file. For locations and monster drops, I will make/alter groups that draw from those generic groups. Thus, huge item groups like Kitchen (with 350+ items!) will get condensed down into a composite of other groups, such as kitchen_tools + imperishable_food + local_fruit + irradiated_fruit, etc. That will make it easier to eyeball and adjust for balance. Plus, when someone adds a new type of fruit, it will automatically show up in the kitchen list without having to add it in manually.

### CODE REVIEW

**Comment by SunshineDistillery** [comment]
I'm starting by adding the new folder and files. 

"Main" will contain generic-but-focused groups that are meant to be used anywhere, groups like pistol, smg, shotgun, rifle. (It will also have some composite groups,like allguns which is just a mix of the above.)

"Locations" will have groups for mapgen features, like anthills and labs. It will also have groups for generic-ish places/furniture. Like office_breakroom is general enough it could be used over and over in many differnt mapgen files. No need to recreate a similar group for each maptile. Very specific item groups that are unlikely to be reused, like hatstore_accessories, will be moved to the appropriate mapgen file instead.

"Map_extras" will have mostly what it does now. Map extras are different enough to deserve their own file instead of trying to mix them in with locations. (not adding it just yet though)

"Gear" will have sets of clothing and carried equipment. These already exist for things like Fireman and Survivor Zeds and they're great. They need their own file because they're different and very long/complicated. Like the main file, these are groups that will be used all over, to stock places, vehicles, map extras, but mainly for zed death drops.

"Monster_drops" A lot of monster drops are still in the item_groups file, so I will move them out. I am also updating a few monsters to drop items more the way that survivorzeds and fireman zombies do, since it's much better. Like the Vault Wanderer, instead of dropping a single item from a list, will drop damaged shoes, pants, shirts, etc. like other zeds do currently.

File: data/json/itemgroups/main.json [hunk 1]
```diff
@@ -0,0 +1,227 @@
+[
+  {
+    "type" : "item_group",
+    "id" : "trader_avoid",
```
Comment (Coolthulhu) [inline data/json/itemgroups/main.json]: This group is currently specified twice: here and in the old, big itemgroups.json

File: data/json/itemgroups/main.json [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+      ["rx12_injector", 50],
+      ["stimpack_ammo", 100],
+      ["survivormap", 50],
+      ["survnote", 10],
+      ["tool_rdx_charge", 10],
+      ["v29", 10]
+    ]
+  },{
+    "type" : "item_group",
+    "id" : "player_crafted",
```
Comment (Coolthulhu) [inline data/json/itemgroups/main.json]: This isn't used anywhere.
I don't really see how would it be used. To produce errors when someone places an item from this group in other group?

**Comment by Coolthulhu** [comment]
The idea for the split is good.

Not sure what will happen with the duplicate group, so I'll wait for the fix.

File: data/json/itemgroups/main.json (hunk 2 above)
Comment (SunshineDistillery) [inline data/json/itemgroups/main.json]: It's like the rare group. It's just an FYI to remind people not to place these items anywhere. It's not important though.  
I can either remove it or make the comment more explicit.

**Comment by Zireael07** [comment]
Great, waiting for the next PR in the series, @SunshineDistillery!

**Pull request merged**